code/game/objects/effects/contraband.dm

```diff
@@ -115,15 +125,36 @@
 		return
 	if(ruined)
 		return
+
 	visible_message(span_notice("[user] rips [src] in a single, decisive motion!") )
 	playsound(src.loc, 'sound/items/poster_ripped.ogg', 100, TRUE)
+	spring_trap(user)
 
 	var/obj/structure/sign/poster/ripped/R = new(loc)
 	R.pixel_y = pixel_y
 	R.pixel_x = pixel_x
 	R.add_fingerprint(user)
 	qdel(src)
 
+/obj/structure/sign/poster/proc/spring_trap(mob/user)
+	if (!trap)
+		return
+
+	to_chat(user, span_warning("There's something sharp behind this! What the hell?"))
+	if(!can_embed_trap(user) || !trap.tryEmbed(user.get_active_hand(), TRUE))
+		visible_message(span_notice("A [trap.name] falls from behind the poster.") )
+		trap.forceMove(user.drop_location())
+	else
+		trap_succeeded()
+
+/obj/structure/sign/poster/proc/trap_succeeded(mob/user)
+
+/obj/structure/sign/poster/proc/can_embed_trap(mob/living/carbon/human/user)
+	if (!istype(user))
+		return FALSE
+
+	return (!user.gloves && !HAS_TRAIT(user, TRAIT_PIERCEIMMUNE))
```

Comment:
why have this bit pulled out? cleanliness?

Reply:
Part neatness but probably more just that the linter at my job complains if there's no newline after every return and it's hard to get out of the habit.